driver-core/src/main/com/mongodb/connection/SocketSettings.java

```diff
@@ -260,12 +264,7 @@ public boolean equals(final Object o) {
 
     @Override
     public int hashCode() {
-        int result = (int) (connectTimeoutMS ^ (connectTimeoutMS >>> 32));
-        result = 31 * result + (int) (readTimeoutMS ^ (readTimeoutMS >>> 32));
-        result = 31 * result + receiveBufferSize;
-        result = 31 * result + sendBufferSize;
-        result = 31 * result + proxySettings.hashCode();
-        return result;
```

Comment:
The bit shifting in the old code became irrelevant, i.e., `hashCode` had to be updated. I updated it to use `Objects.hash`.